File: 645. Set Mismatch/eric.go
func findErrorNums(nums []int) []int {
    mp := make(map[int]int)
    ans := make([]int, 2)
    var val int
    var ok bool
    
    for i := 0; i < len(nums); i++ {
        mp[nums[i]]++
    }

    for i := 1; i <= len(nums); i++ {
        val, ok = mp[i];

        if !ok {
            ans[1] = i
        }

        if val > 1 {
            ans[0] = i
        }
    }

    return ans
} 

File: 645. Set Mismatch/eric.cpp
class Solution {
public:
    vector<int> findErrorNums(vector<int>& nums) {
        vector<int> ans (2, 0);
        unordered_map<int, int> mp;

        for (auto num: nums) {
            mp[num]++;
        }

        for (int i = 1; i <= nums.size(); i++) {
            if (mp.count(i) == 0) {
                ans[1] = i;
            }

            if (mp[i] > 1) {
                ans[0] = i;
            }
        }

        return ans;
    }
};
